src/conditions.c

```diff
@@ -0,0 +1,18 @@
+#include "vctrs.h"
+#include "utils.h"
+
+
+void stop_scalar_type(SEXP x, const char* arg_str) {
+  SEXP arg;
+  if (!strlen(arg_str)) {
+    arg = PROTECT(Rf_mkString(arg_str));
+  } else {
+    arg = PROTECT(R_NilValue);
```

Comment:
That doesn't actually need protecting, right?

Reply:
Right, this is just to avoid counted protection.

Reply:
It's just a bit weird since you don't unprotect (because of the error)

Reply:
ah yes, not protecting would cause a protection balance warning in rchk. And I didn't unprotect partly because it's weird to add unreachable code, and partly because I wanted to test whether rchk understands the `error` call.